app/models/concerns/labware_creators/donor_pooling_calculator.rb

```diff
@@ -261,18 +288,29 @@ def calculate_chip_loading_volume(num_cells_per_chip_well)
     num_cells_per_chip_well / chip_loading_conc
   end
 
-  # This method calculates the full allowance volume for a given chip loading volume.
+  # This method calculates the allowance volume for a given chip loading volume and allowance band.
   # It uses configuration values from the scrna_config for the desired
   # number of runs, the volume taken for cell counting, and the wastage volume.
   #
   # @param chip_loading_volume [Float] the chip loading volume
-  # @return [Float] the full allowance volume
-  def calculate_full_allowance(chip_loading_volume)
+  # @return [Float] the allowance volume
+  def calculate_allowance(chip_loading_volume, allowance_band) # rubocop:disable Metrics/AbcSize
     scrna_config = Rails.application.config.scrna_config
 
-    (chip_loading_volume * scrna_config[:desired_number_of_runs]) +
-      (scrna_config[:desired_number_of_runs] * scrna_config[:volume_taken_for_cell_counting]) +
-      scrna_config[:wastage_volume]
+    case allowance_band
+    when 'Full allowance'
+      (chip_loading_volume * scrna_config[:desired_number_of_runs]) +
+        (scrna_config[:desired_number_of_runs] * scrna_config[:volume_taken_for_cell_counting]) +
+        scrna_config[:wastage_volume]
+    when '2 pool attempts, 1 count'
+      (chip_loading_volume * scrna_config[:desired_number_of_runs]) + scrna_config[:volume_taken_for_cell_counting] +
+        scrna_config[:wastage_volume]
+    when '1 pool attempt, 2 counts'
+      chip_loading_volume + (scrna_config[:desired_number_of_runs] * scrna_config[:volume_taken_for_cell_counting]) +
+        scrna_config[:wastage_volume]
+    when '1 pool attempt, 1 count'
+      chip_loading_volume + scrna_config[:volume_taken_for_cell_counting] + scrna_config[:wastage_volume]
+    end
```

Comment:
I've asked the same of Sabrine in her story - can we get rid of the `desired_number_of_runs` constant please, and just hardcode `2` instead? It's misleading, as it isn't necessarily the desired number of runs - the customer might have only supplied enough for 1 run, and that's OK.

Reply:
1. Yeah happy to make that changes, feels more consistent.
2. Good suggestion, an enum makes sense.